x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/GetUserPrivilegesResponse.java

```diff
@@ -104,12 +131,34 @@ public boolean equals(Object other) {
             && Objects.equals(configurableClusterPrivileges, that.configurableClusterPrivileges)
             && Objects.equals(index, that.index)
             && Objects.equals(application, that.application)
-            && Objects.equals(runAs, that.runAs);
+            && Objects.equals(runAs, that.runAs)
+            && Objects.equals(remoteIndex, that.remoteIndex);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(cluster, configurableClusterPrivileges, index, application, runAs);
+        return Objects.hash(cluster, configurableClusterPrivileges, index, application, runAs, remoteIndex);
+    }
+
+    public record RemoteIndices(Indices indices, Set<String> remoteClusters) implements ToXContentObject, Writeable {
```

Comment:
Nit: Rename to `remoteIndices`? Can we avoid using same `indices` variable name in both RemoteIndices and Indices?

Reply:
I think `indices` makes sense here because it's an actual `Indices` instance, which `RemoteIndices` uses as a field. Renaming to `remoteIndices` would be kind of quirky because we'd have code like 

```
RemoteIndices remoteIndices = new RemoteIndices(); 
remoteIndices.remoteIndices();
```

Reply:
I am OK with that quirk. For example, it makes find easier when using non-IDEs like GitHub PRs.

If we implemented from scratch, perhaps we would have differentiated the two as LocalIndices versus RemoteIndices. The former has an implicit clusterAlias, the latter has explicit clusterAlias(es). The varable names might reflect that difference as LocalIndices.localIndices versus RemoteIndices.remoteIndices.

This is a nit. I will leave the decision to you. If you don't want to change it, that is fine. Just wanted you to consider if there is a way to make find easier in non-IDEs by possibly using unique variable names.

Reply:
Sounds good! Think it's one of those things where it comes down more to personal preference than anything. I'll run with `indices` to keep things moving. Since this is just a field name, as opposed to part of the REST API, we can rename if naming inspiration strikes in the future.